torch/csrc/distributed/rpc/rref.cpp

```diff
@@ -136,16 +136,7 @@ UserRRef<T>::UserRRef(
 
 template <typename T>
 UserRRef<T>::~UserRRef() {
-  // TODO: queue this in RRefContext instead of doing it here.
-  auto& ctx = RRefContext::getInstance();
-  if (ctx.getWorkerId() != ownerId_) {
-    auto fm = ctx.agent()->send(
-        ctx.agent()->getWorkerInfo(ownerId_),
-        RRefUserDelete(rrefId_, forkId_).toMessage());
-
-    fm->addCallback(
-        [](const Message& message) { RRefContext::handleException(message); });
-  }
+  RRefContext::getInstance().delUser(ownerId_, rrefId_, forkId_);
```

Comment:
Can we catch and log exceptions here? Would be pretty risky to throw exceptions from the destructor.